```
ccache /opt/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools/bin/xt-xcc    CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr_prebuilt.elf -T linker.cmd -Wl,-Map=/home/nashif/Work/github/zephyr/build/zephyr/zephyr.map -u_OffsetAbsSyms -u_ConfigAbsSyms -Wl,--whole-archive ../libapp.a libzephyr.a arch/xtensa/core/startup/libarch__xtensa__core__startup.a lib/libc/minimal/liblib__libc__minimal.a soc/xtensa/intel_s1000/libsoc__xtensa__intel_s1000.a subsys/usb/class/hid/libsubsys__usb__class__hid.a drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive kernel/libkernel.a CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -L/home/nashif/Work/github/zephyr/build/zephyr -lgcc -lhal -nostdlib -static -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none
libzephyr.a(xt_zephyr.S.obj):(.literal+0x8): undefined reference to `_timer_int_handler'
make[2]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:98: zephyr/zephyr_prebuilt.elf] Error 2
make[2]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make[1]: *** [CMakeFiles/Makefile2:426: zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2
make[1]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make: *** [Makefile:87: all] Error 2
make: Leaving directory '/home/nashif/Work/github/zephyr/build'
```

--- a/drivers/timer/xtensa_sys_timer.c
+++ b/drivers/timer/xtensa_sys_timer.c
@@ -57,6 +57,17 @@ static void ccompare_isr(void *arg)
 	z_clock_announce(dticks);
 }
 
+/* The legacy Xtensa platform code handles the timer interrupt via a
+ * special path and must find it via this name.  Remove once ASM2 is
+ * pervasive.
+ */
+#ifndef CONFIG_XTENSA_ASM2
+void _timer_int_handler(void *arg)
+{
+	return ccompare_isr(arg);
+}
+#endif
+
 int z_clock_driver_init(struct device *device)
 {
 	IRQ_CONNECT(TIMER_IRQ, 0, ccompare_isr, 0, 0);

